File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
class Solution {
    List<Integer> result = new ArrayList<>();

    public List<Integer> postorderTraversal(TreeNode root) {
        if(root == null) return result;
        
        postorderTraversal(root.left);
        postorderTraversal(root.right);
        result.add(root.val);
        
        return result;
    }
}

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cs
public class Solution {
    IList<int> result = new List<int>();

    public IList<int> PostorderTraversal(TreeNode root) {
        if(root == null) return result;
        
        PostorderTraversal(root.left);
        PostorderTraversal(root.right);
        result.Add(root.val);
        
        return result;
    }
}

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */